python/cuml/dask/neighbors/nearest_neighbors.py

```diff
@@ -27,6 +27,9 @@
 from uuid import uuid1
 
 
+DEFAULT_BATCH_SIZE = 2000000
```

Comment:
Is it correct that if the batch size is now 2e6, each individual Dask task will predict up to 2 million rows. And previously, with the batch size of 1024, each Dask task would predict up to 1024 rows at once?

Reply:
The batch size was always intended to be 2e6. For some reason, this was being set to 1024 only in the dask layer, which was likely done inadvertently (and was probably my fault). 

Here's where it's defaulting to 2e6 in the underlying NearestNeighborsMG,pyx class: https://github.com/rapidsai/cuml/blob/branch-0.17/python/cuml/neighbors/nearest_neighbors_mg.pyx#L153

Reply:
Yes indeed, the range of possible values is very large. In one case (low `batch_size`) there's a very low idle time but relatively low GPU usage, in the other case (high `batch_size`) a longer transfer time is expected before any work start, but then GPU is used at full capacity. The optimal value depend on the GPU and the communications setups, as well as the data that is used.

Reply:
Transfers are asynchronous and using UCX, so fast Ethernet or Infiniband *should* both be fast.  @beckernick, the additional batching was added for users to fine-tune the behavior of nearest neighbors without having rechunk their Dask arrays. We set the default batch size to be large so that users can rechunk their Dask arrays if out of memory errors occur. The additional batching allows more fine-grained control without the need for this additional chunking. 2M was chosen as a reasonable middle ground- with 500 features, this would require about 4GB of additional overhead on each index partition.